prov/lnx/src/lnx_ops.c

```diff
@@ -374,14 +416,15 @@ static int lnx_process_recv(struct lnx_ep *lep, struct iovec *iov, void **desc,
 	/* nothing on the unexpected queue, then allocate one and put it on
 	 * the receive queue
 	 */
-	rx_entry = get_rx_entry(NULL, iov, desc, count, addr, tag, ignore,
-				context, lnx_ep_rx_flags(lep));
-	rx_entry->rx_entry.msg_size = ofi_total_iov_len(iov, count);
+	rx_entry = get_rx_entry(NULL, iov, (desc) ? &desc : NULL, count,
+				match_attr.lm_addr, tag, ignore, context,
+				flags);
 	if (!rx_entry) {
 		rc = -FI_ENOMEM;
 		goto out;
 	}
-	rx_entry->rx_peer = lp;
+	if (iov)
+		rx_entry->rx_entry.msg_size = ofi_total_iov_len(iov, count);
```

Comment:
I don't think this is necessary since the msg size refers to the total size of the incoming message (provided by the peer), not the total available size of the posted receive.

Reply:
done